server/controllers/payment.js

```diff
@@ -0,0 +1,190 @@
+const User = require("../models/User");
+const StripeServices = require("../services/StripePayment");
+const asyncHandler = require("express-async-handler");
+const {
+  stripePayment,
+  createCustomer,
+  getCustomer,
+  subscriptionOneTimePayment,
+  createIntent,
+  getPaymentMethods,
+} = StripeServices;
+
+// @route POST /:id/single-pay
+// @desc create single payment checkout session
+// @access Private
+exports.payment = asyncHandler(async (req, res, next) => {
+  const id = req.user.id;
+  if (id === req.params.id) {
+    await stripePayment(req, res, function (err) {
+      if (err) {
+        return res.json({
+          success: false,
+          errors: {
+            title: "Stripe payment failed",
+            detail: err.message,
+            error: err,
+          },
+        });
+      }
+    });
```

Comment:
Decorating your function with `async` keyword allows you to use `await` and assign the return value of a Promise to a variable instead of passing a callback.

Reply:
👍🏻Done